### ISSUE TITLE
Folder list does not get focus

### ISSUE THREAD

**Issue by sma-ops** [kewisch/quickmove-extension#103]
When opening the quick move folder list dialog via hot key, it only sometimes has the keyboard focus. The very convenient feature of typing the first letter(s) of a folder and move via enter becomes actually a nuisance, because the typed letters instead are processed in the email list and lead to all kinds of operations on the messages list (depending on which letters to typed).

TB: 115.2.3
OS: Ubuntu 22.04.3 LTS

**Comment by neliorc** [comment]
Happens on Windows as well. My understanding is that the new popup takes a while to render, and if you start typing before it fully renders, the keypresses go to Thunderbird and not the popup. So now I always wait a second or so before starting to type, which is annoying.

**Comment by Cito** [comment]
I think #89 is/was a related issue.

**Comment by neliorc** [comment]
I think you're right. Did some testing and making sure the mouse was never on the popup seems to have solved the issue and I don't need it to fully render to start typing.

**Comment by neliorc** [comment]
Actually, what I described still happens. While the popup is being loaded, key presses are sent to Thunderbird. The previous popup was much simpler and loaded instantly, while the new one takes a while (a second or so). It may seem quick, but it makes it nearly impossible to Shift + M and start typing the folder name immediately, as I did before.

**Comment by IGZmanuelMartinVivaldi** [comment]
In my case (Win11) I noticed it lost the focus only when the new popup is under the mouse. When the popup window expand and a new folder row is under the mouse, the folder row get the focus and the input lost it.

**Comment by kewisch** [comment]
#89 is fixed and will be released soon. Unfortunately I'm not sure if I can speed up the timing for the popup. Part of this would be Thunderbird maybe needing some machinery to capture keyboard input while the popup is opening and forward that (unlikely due to complexity and edge case). The other part may be loading the popup quicker, but I can only do that as fast as I get the data from the API.

With a little effort someone might be able to sit down and see if the folder loading can be deferred in a way that popup load time goes down, but this will also be an edge case. I'd suggest to train some milliseconds delay between the shortcut and continue typing. Not ideal, but I don't really have a better idea at the moment. Ideas welcome!

**Comment by sma-ops** [comment]
thank you, the issue disappeared for me with Quick Folder Move v2.7.0 on TB 115.3.1
